fds/services/git_service.py

```diff
@@ -55,3 +55,9 @@ def add(self, add_argument: str) -> Any:
 
     def commit(self, message: str) -> Any:
         execute_command(["git", "commit", "-am", message], capture_output=False)
+
+    def clone(self, url: str) -> Any:
+        execute_command(["git", "clone", url], capture_output=False)
+        # Todo: Also support getting the directory to clone from the user
+        git_repo_name = url.split("/")[-1]
+        return git_repo_name.split(".git")[0]
```

Comment:
I would factor this logic out to a pure function and do unit tests on it.
What will happen on SSH repo URLs?

Reply:
I thought about it, and I tested, it works for ssh too, now added tests for both cases separately